lib/rtp_h264/depayloader.ex

```diff
@@ -0,0 +1,117 @@
+defmodule Membrane.Element.RTP.H264.Depayloader do
+  @moduledoc """
+  Depayloads H264 RTP payloads into H264 NAL Units.
+  """
+  use Membrane.Element.Base.Filter
+  use Membrane.Log
+
+  alias Membrane.Buffer
+  alias Membrane.Caps.{RTP, Video.H264}
+  alias Membrane.Event.Discontinuity
+  alias Membrane.Element.RTP.H264.{FU, NAL, StapA}
+
+  @frame_prefix <<1::32>>
+  @type sequence_number :: 0..65_535
+
+  def_output_pads output: [
+                    caps: {H264, stream_format: :byte_stream}
+                  ]
+
+  def_input_pads input: [
+                   caps: {RTP, payload_type: :dynamic},
+                   demand_unit: :buffers
+                 ]
+
+  defmodule State do
+    @moduledoc false
+    defstruct parser_acc: nil
+  end
+
+  def handle_init(_) do
+    {:ok, %State{}}
+  end
+
+  @impl true
+  def handle_caps(:input, _caps, _context, state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_process(_pad, %Buffer{payload: payload} = buffer, _ctx, state) do
+    with {:ok, {header, rest}} <- NAL.Header.parse_unit_header(payload),
+         {{:ok, _}, _} = action <-
+           (case NAL.Header.decode_type(header) do
+              :single_nalu ->
+                buffer_output(payload, buffer)
```

Comment:
no need to pass both payload and buffer, as the payload can be taken from the buffer